### ISSUE TITLE
Missing AWS Resources

### ISSUE THREAD

**Issue by jamesgoodhouse** [sparkleformation/sparkle_formation#200]
I'm currently trying to use SFN to create an ELBv2 as well as manage Certificates in AWS. It appears that neither of these are currently defined within SFN. I figured this wouldn't be an issue, as I can create my own "dynamic" resource. Unfortunately this isn't an ideal solution as the update plan appears to have no idea what changes will be occurring when I update a stack.

Is there a way to add additional AWS resources somehow? I have scoured the documentation and have not found anything. I see in the codebase where the JSON definition is for the AWS resources, as well as some additional code/logic for how to handle various update scenarios. Is there a way I can update this and load it into SFN, or must it be part of the gem itself?

**Comment by jimcroft** [comment]
[AWS::ElasticLoadBalancingV2::LoadBalancer](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-elasticloadbalancingv2-loadbalancer.html) works fine for us in SF. What issues or errors are you seeing?

**Comment by jamesgoodhouse** [comment]
@jimcroft: How are you implementing it, as I don't believe SF has built in support for V2 ELBs. The only way I could get it to work was just making a custom resource, but then you lose the benefit of knowing what is going to change when you make updates.

We ultimately decided to stick with CloudFormation, with the plan to switch over to Terraform, as it seems that SF is not longer under active development.

**Comment by webframp** [comment]
Pretty sure @chrisroberts is planning an update eventually now that schemas are being published (#197) but in the meantime just declare the type on the resource instead of using the builtin `dynamic!` helper method:
```rb
your_v2_lb do
  type 'AWS::ElasticLoadBalancingV2::LoadBalancer'
  properties do
  #... properties
  end
end
```

**Comment by chrisroberts** [comment]
The builtin resource list was updated in the last release, and are updated as releases occur. The builtin resources are not a requirement for sparkleformation and are included simply as a convenience for quickly inserting resource definitions. If a resource is not available in the builtin list, it can be easily stubbed using a dynamic. For example, using the v2 ELB resource. The dynamic should be used in a template with a call like:

```ruby
dynamic!(:my_elb, :elastic_load_balancing_v2_load_balancer)
```

Because (in our example case) the resource type isn't included in the builtin resource list, it can be stubbed out as a dynamic. This would look like:

```ruby
SparkleFormation.dynamic(:elastic_load_balancing_v2_load_balancer) do |name|
  resources.set!("#{name}_elastic_load_balancing_v2_load_balancer".to_sym) do
    type 'AWS::ElasticLoadBalancingV2::LoadBalancer"
  end
end
```

This creates a "stub resource" and can be removed (if desired) once the resource type shows up in the builtin list.

Cheers!
